src/zarr/core/_info.py

```diff
@@ -109,18 +109,19 @@ def __repr__(self) -> str:
         Read-only          : {_read_only}
         Store type         : {_store_type}""")
 
-        kwargs = dataclasses.asdict(self)
+        # We can't use dataclasses.asdict, because we only want a shallow dict
+        kwargs = {field.name: getattr(self, field.name) for field in dataclasses.fields(self)}
+
         if self._chunk_shape is None:
             # for non-regular chunk grids
             kwargs["chunk_shape"] = "<variable>"
-        if self._compressor is not None:
-            template += "\nCompressor         : {_compressor}"
 
-        if self._filters is not None:
+        if len(self._filters) > 0:
```

Comment:
Will this prevent arrays with no filters at all from displaying the fact that they have no filters via the `.info` method?

Reply:
likely out of scope, since my question is really about the pre-existing behavior of this function

Reply:
The "Filters     : ... " line wasn't shown if `filters` was `None`, but it was shown if `filters` was `()`. Now, it is not shown if `None` or `()`.

Reply:
Don't we want to explicitly show the value of `filters`, even if it's `()`?

Reply:
We could. I don't have a strong preference here.

Reply:
IMO it's better to show a static set of properties with this function, but if that's annoying to implement here we can defer for a separate PR

Reply:
What about "Serializer", which doesn't exist for v2 arrays?

Reply:
maybe `None` there? in principle there's nothing stopping someone from implementing a v2-compatible `BytesCodec` in numcodecs, in which case we might end up with v2 arrays with a `serializer`

Reply:
Well, there is no place in the v2 metadata for a serializer.
I opted for not showing the serializer field in the Array.info.